src/SfxWeb/src/app/Models/eventstore/rcaEngine.ts

```diff
@@ -170,3 +167,22 @@ export const getSimultaneousEventsForEvent = (configs: IConcurrentEventsConfig[]
 
   return simulEvents;
 }
+
+export const getPeriodicEvent = (configs: IDiffAnalysis[], inputEvents: IRCAItem[]): IAnalysisResultDiff[] => {
+  const results = configs.map(config => {
+    return {
+      config,
+      events: []
+    }
+  })
+
+  inputEvents.forEach(event => {
+    results.forEach(config => {
+      if(config.config.eventType === event.kind && validMappings(event, event, config.config)) {
```

Comment:
I'm sure there's a good reason for it, but I don't quite understand why validMappings needed to be called with the source and target as the same event